tutor/src/components/enroll/index.cjsx

```diff
@@ -0,0 +1,90 @@
+React = require 'react'
+_ = require 'underscore'
+{Promise} = require 'es6-promise'
+
+{CourseEnrollmentActions, CourseEnrollmentStore} = require '../../flux/course-enrollment'
+{CourseActions, CourseStore} = require '../../flux/course'
+ENTER = 'Enter'
+
+ConfirmJoin = require './confirm-join'
+JoinConflict = require './join-conflict'
+Router = require '../../helpers/router'
+
+{MessageList} = require 'shared'
+
+Enroll = React.createClass
+
+  contextTypes:
+    router: React.PropTypes.object
+
+  componentWillMount: ->
+    {enrollmentCode} = Router.currentParams()
+    CourseEnrollmentStore.on('changed', @onCourseEnrollmentChange)
+    CourseEnrollmentActions.create(enrollmentCode)
+
+  componentWillUnmount: ->
+    CourseEnrollmentStore.off('changed', @onCourseEnrollmentChange)
+
+  redirectToDashboard: ->
+    @context.router.transitionTo(
+      Router.makePathname('dashboard', {courseId: CourseEnrollmentStore.getCourseId()})
+    )
+
+  # Wait for the course to load and wait at least 1.5 secs
+  # so our success message is briefly displayed, then redirect to dashboard
+  onComplete: ->
+    loadCourse = new Promise((resolve, reject) ->
+      CourseStore.once('course.loaded', resolve)
+      CourseActions.load(CourseEnrollmentStore.getCourseId())
+    )
+    successTimer = new Promise((resolve, reject) -> _.delay(resolve, 1500))
+    Promise.all([loadCourse, successTimer]).then(@redirectToDashboard)
+
```

Comment:
Thanks yeah took me a while to figure out how to load the course while the timer was going, but seems to work great.